lib/std/os/linux.zig

```diff
@@ -6113,12 +6113,13 @@ pub const POSIX_FADV = switch (native_arch) {
 };
 
 /// The timespec struct used by the kernel.
-pub const kernel_timespec = if (@sizeOf(usize) >= 8) timespec else extern struct {
+pub const kernel_timespec = extern struct {
     tv_sec: i64,
     tv_nsec: i64,
 };
 
-pub const timespec = extern struct {
+// TODO: This is a temporary hack until we figure out a coherent Y2038 strategy.
```

Comment:
https://github.com/ziglang/zig/issues/4726#issuecomment-2190337877

I think I might try tackling this in a future PR. This hack should work for now though.

Reply:
It's unclear what exactly makes this a "temporary hack". Request: instead of a TODO comment using the words "temporary hack", explain specifically the limitations of the code, why it is not adequate in its current form. Then make note that there is a follow-up issue in the PR description, and then when it is merged, file the follow-up issue. Related: #363

Reply:
The follow-up issue is basically #4726 already. (See my comment there, by the way - there's a question that needs answering before I can proceed with trying to address that issue later on.)

Would this be descriptive enough?

```
// https://github.com/ziglang/zig/issues/4726#issuecomment-2190337877
//
// It is not technically correct to define timespec as equal to kernel_timespec on riscv32. In the
// UAPI headers, they are still distinct structures. But until #4726 is fully addressed, this
// temporary workaround will allow all the syscall wrappers in this module to work correctly.
//
// The reason for doing this is that all the existing wrappers use timespec. This is wrong for
// riscv32 which uses a newer time ABI where even the old syscalls like ppoll use kernel_timespec
// which is 64-bit. Contrast with older kernel ports where the ABI requires you to use the separate
// ppoll_time64 syscall for 64-bit time.
```

Reply:
That's great, thank you. Even the link alone is sufficient to address my feedback.